### PR TITLE
#4085 [Manager] Creating a workspace fails too frequently

### PR DESCRIPTION
**Pull request by eechih**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #4085 )
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/components/Workspace/Create/WorkspaceQuick.js [hunk 1]
```diff
@@ -305,6 +305,10 @@ const WorkspaceQuick = props => {
       setActiveStep(0);
       setFiles([]);
     } catch (e) {
+      eventLog.error({
+        title: `Failed to creates workspace ${workspaceName}.`,
+        message: `If you want any help, please go to the official website https://github.com/oharastream/ohara/issues to create an issue.`,
```
Comment (eechih) [inline ohara-manager/client/src/components/Workspace/Create/WorkspaceQuick.js]: 假如建立 Workspace 失敗，提醒使用者前往 https://github.com/oharastream/ohara/issues 提問

File: ohara-manager/client/src/api/utils/waitUtils.js [hunk 2]
```diff
@@ -14,22 +14,15 @@
...
 
-import { isEmpty } from 'lodash';
+import { get, includes, isEmpty } from 'lodash';
 
-export const waitForConnectReady = res => {
-  const { classInfos } = res.data.result;
-  return !isEmpty(classInfos);
-};
+export const waitForConnectReady = res =>
+  !isEmpty(get(res, 'data.result.classInfos'));
```
Comment (eechih) [inline ohara-manager/client/src/api/utils/waitUtils.js]: @saivirtue 建立 Workspace 常常遇到失敗的原因分析如下，在整過建立過程中，最主要也是最重要的一個步驟就是確認 worker 是否啟動成功，這邊使用 wait 的 checkFn 不斷嘗試打 API 進行確認。如果遇到 API 回應太慢（超過 20s）前端的 HTTP 框架會報錯，報錯後回傳的物件沒有包含 result，因此 checkFun 執行後會發生 error，而這個 error 沒有被捕捉也沒有反應在 UI 上。

File: ohara-manager/client/src/api/waitApi.js [hunk 3]
```diff
@@ -27,19 +28,21 @@ const wait = async params => {
     sleep = 2000,
   } = params;
   const res = await axiosInstance.get(url);
-  if (checkFn(res, paramRes)) {
+  if (get(res, 'data.isSuccess') === true && checkFn(res, paramRes) === true) {
```
Comment (saivirtue) [inline ohara-manager/client/src/api/waitApi.js]: 這裡需要比對布林值嗎?

Comment (eechih) [reply ohara-manager/client/src/api/waitApi.js]: 嚴格比對。避免其他 “非 false” 的回傳值

Comment (saivirtue) [reply ohara-manager/client/src/api/waitApi.js]: I agree. But we use `if(res.data.isSuccess)` in multiple files (ex: brokerAPi.js, connectorApi.js...)
Should we apply this rule to other files?

File: ohara-manager/client/src/components/Workspace/Create/WorkspaceQuick.js (hunk 1 above)
Comment (konekoya) [inline ohara-manager/client/src/components/Workspace/Create/WorkspaceQuick.js]: Would rephrase this to:
_Oops, it seems like our UI is not working as it should. For more information or reporting the bug, please go to
https://github.com/oharastream/ohara/issues_

:)

File: ohara-manager/client/src/api/waitApi.js (hunk 3 above)
Comment (eechih) [reply ohara-manager/client/src/api/waitApi.js]: 這邊會使用嚴格比對主要是針對 checkFn，isSuccess 只是順便而已，因為我們要限制 checkFn 一定回傳 boolean 而不是其他型態，而 isSuccess 的回傳值已經肯定是 boolean，所以不用嚴格比對。我會調整成 if (get(res, 'data.isSuccess') && checkFn(res, paramRes) === true)

Comment (eechih) [reply ohara-manager/client/src/api/waitApi.js]: @saivirtue any comment?

Comment (saivirtue) [reply ohara-manager/client/src/api/waitApi.js]: noop. Thanks for the explanation :)
